source/common/memory/stats.h

```diff
@@ -51,5 +66,84 @@ class Stats {
   static void dumpStatsToLog();
 };
 
+class AllocatorManager {
+public:
+  AllocatorManager(Api::Api& api, Envoy::Stats::Scope& scope,
+                   const envoy::config::bootstrap::v3::MemoryAllocatorManager& config)
+      : bytes_to_release_(config.bytes_to_release()),
+        memory_release_interval_msec_(std::chrono::milliseconds(
+            PROTOBUF_GET_MS_OR_DEFAULT(config, memory_release_interval, 1000))),
+        allocator_manager_stats_(MemoryAllocatorManagerStats{
+            MEMORY_ALLOCATOR_MANAGER_STATS(POOL_COUNTER_PREFIX(scope, "tcmalloc."))}) {
+    configureBackgroundMemoryRelease(api);
+  };
+
+  ~AllocatorManager() {
+    {
+      Thread::LockGuard guard(mutex_);
+      terminating_ = true;
```

Comment:
Think we can just rely on the dispatcher synchronization e.g. the call to exit here should cause the dispatcher to exit even if the timer is queued, which should simplfy the logic getting rid of terminating_ bool and the associated mutex